app/src/main/java/com/cornellappdev/android/eatery/ui/components/general/AppStoreRatingPopup.kt

```diff
@@ -0,0 +1,214 @@
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.draw.shadow
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.window.Dialog
+import androidx.core.content.ContextCompat.startActivity
+import com.cornellappdev.android.eatery.ui.components.home.EateryDetailLoadingScreen
+import com.cornellappdev.android.eatery.ui.theme.EateryBlue
+import com.cornellappdev.android.eatery.ui.theme.EateryBlueTypography
+import com.cornellappdev.android.eatery.ui.theme.GrayZero
+import com.cornellappdev.android.eatery.util.AppStorePopupRepository
+import com.cornellappdev.android.eatery.util.appStorePopupRepository
+import com.valentinilk.shimmer.ShimmerBounds
+import com.valentinilk.shimmer.rememberShimmer
+
+@Composable
+fun AppStoreRatingPopup(
+    navigateToSupport: () -> Unit,
+    appStorePopupRepository: AppStorePopupRepository = appStorePopupRepository()
+) {
+    val showPopup = appStorePopupRepository.popupShowing.collectAsState().value
+    if (showPopup) {
+        Dialog(appStorePopupRepository::dismissPopup) {
+            AppStoreRatingDialog(navigateToSupport, appStorePopupRepository::dismissPopup)
+        }
+    }
+}
+
+@OptIn(ExperimentalAnimationApi::class)
+@Composable
+private fun AppStoreRatingDialog(navigateToSupport: () -> Unit, onDismiss: () -> Unit) {
+    var rating by remember { mutableStateOf(0) }
+    val context = LocalContext.current
+    val packageName = context.packageName
+
+    AnimatedContent(targetState = rating) { currentRating ->
+        when (currentRating) {
+            0 -> RatingPrompt(rating, onChangeRating = { rating = it }, onDismiss = onDismiss)
+
+            5 -> ActionPrompt(
+                "Awesome! We'd love to hear more in a review",
+                "Open PlayStore",
+                onButtonPress = {
+                    try {
+                        startActivity(
+                            context, Intent(
+                                Intent.ACTION_VIEW,
+                                Uri.parse("market://details?id=$packageName")
+                            ), null
+                        )
+                    } catch (e: ActivityNotFoundException) {
+                        startActivity(
+                            context,
+                            Intent(
+                                Intent.ACTION_VIEW,
+                                Uri.parse("https://play.google.com/store/apps/details?id=$packageName")
+                            ),
+                            null
+                        )
+                    } finally {
+                        onDismiss()
```

Comment:
I saw that in your demo this opened the eatery app in a web browser instead of the play store app. Is this because that android phone you tested on didn't have Google Play? 

I just tested and it seems like it indeed opens the play store correctly on my test phone :-)

Reply:
Yes it was only bc my emulator did not have Google Play installed.